File: fixtures/scale/objc/large/Families/Adler32/Adler32Descriptor.m
#import "Adler32Descriptor.h"
#import "Adler32Cipher.h"
#import "Adler32Vectors.h"

@implementation Adler32Descriptor

- (NSString *)family {
    return @"adler32";
}

- (NSString *)suite {
    return @"hash";
}

- (id<CLCipher>)cipher {
    return [[Adler32Cipher alloc] init];
}

- (NSArray *)vectors {
    return [Adler32Vectors all];
}

@end

File: fixtures/scale/php/large/src/Families/Adler32/Adler32Descriptor.php
<?php

namespace CipherLab\Families\Adler32;

use CipherLab\Abstractions\Cipher;
use CipherLab\Abstractions\FamilyDescriptor;

// Registered once in FamilyCatalog; ties together Adler32Cipher and
// Adler32Vectors under the "hash" suite.
class Adler32Descriptor implements FamilyDescriptor
{
    public function family(): string
    {
        return 'adler32';
    }

    public function suite(): string
    {
        return 'hash';
    }

    public function cipher(): Cipher
    {
        return new Adler32Cipher();
    }

    public function vectors(): array
    {
        return Adler32Vectors::all();
    }
}
